compiler/rustc_mir_transform/src/coroutine.rs

```diff
@@ -1105,11 +1168,25 @@ fn compute_layout<'tcx>(
             // around inside coroutines, so it doesn't matter which variant
```

Comment:
This comment was cut off in the patch, it starts off with:
```
             // Note that if a field is included in multiple variants, we will
            // just use the first one here. That's fine; fields do not move
```

I had started writing this PR comment with a question asking if this code comment needs revision for consistency with this change. After reflecting on things, i *think* the whole picture here remains coherent, but now I want to double check my interpretation.

Before this patch, the upvars were disjoint from the fields carried across yields of a generator/coroutine, so the comment above simply didn't apply to upvars.

After this patch, the upvars are now mapped into the Unresumed variant of a generator/coroutine. But that isn't inconsistent with the comment above ... as long as the Unresumed variant is itself the *first* variant... right?